src/web/components/ApiKeyManagement/KeyItem.tsx

```diff
@@ -18,6 +21,7 @@ function KeyItem({ apiKey: apiKeyInitial, onEdit, onDisable, availableRoles }: K
   const [apiKey, setApiKey] = useState<ApiKeyDTO>(apiKeyInitial);
   const [showKeyDisableDialog, setShowKeyDisableDialog] = useState<boolean>(false);
   const [showKeyEditDialog, setShowKeyEditDialog] = useState<boolean>(false);
+  const [keyNeedsRotating, setKeyNeedsRotating] = useState<boolean>(false);
```

Comment:
maybe `showKeyNeedsRotatingWarning` would be more consistent with the other state variables? Or something similar.

Reply:
Agreed